debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -127,13 +128,23 @@ public R apply(R r) {
             }
         }));
 
+        boolean isDeleteEvent = payload == "" || payload == null;
+
+        Struct updatedValue = (isDeleteEvent && routeTombstoneOnEmptyPayload)
```

Comment:
You could simplify this by having one if condition and setting null in both variables.
Or even treat this as separate code branch, since we shouldn't even process the remaining parts of the envelop, an early return would be more efficient

Reply:
Yeah I think that makes a lot of sense, I’ll make that change!

Reply:
I've done this.